File: back-end/taskManager/Domain/Project.cs
using System.Collections.ObjectModel;
using System.ComponentModel.DataAnnotations;
using task_manager.Domain.Enums;

namespace task_manager.Domain;

public class Project
{
    public Project()
    {
        UserProjects = new Collection<UserProject>();
        Tasks = new Collection<Task>();
    }

    public Guid ProjectId { get; set; }

    [Required]
    [StringLength (300)]
    public string? Name { get; set; }
    public EnumProjectStatus? Status{ get; set; }
    public  DateTime? StartDate { get; set; }
    public DateTime? EndDate { get; set; }
    public ICollection<UserProject>? UserProjects { get; set; }
    public ICollection<Task>? Tasks { get; set; }
}

File: front-end/src/app/shared/Models/Project.ts
import { User } from "./User";

export interface Project
{
  projectId?: string;
  name?: string;
  status?: ProjectStatus;
  startDate?: string;
  endDate?: string;
  users?: User[];
}

export enum ProjectStatus{
  Cancelled = 0,
  Ongoing = 1,
  Delayed = 2,
  Completed = 3
}
